rust/src/delta.rs

```diff
@@ -613,6 +662,143 @@ impl std::fmt::Debug for DeltaTable {
     }
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum DeltaTransactionError {
+    #[error("Version already exists")]
+    VersionExists,
+
+    #[error("Maximum number of retries exceeded for committing the transaction")]
+    CommitRetriesExceeded,
+
+    #[error("RecordBatch is missing partition column in Delta schema.")]
+    MissingPartitionColumn,
+
+    #[error("Commit failed: {source}")]
+    Storage { source: StorageError },
+
+    #[error("DeltaTable interaction failed: {source}")]
+    DeltaTable { 
+        #[from]
+        source: DeltaTableError,
+    },
+
+    #[error("Action serialization failed: {source}")]
+    ActionSerializationFailed {
+        #[from]
+        source: serde_json::Error, 
+    }
+}
+
+impl From<StorageError> for DeltaTransactionError {
+    fn from(error: StorageError) -> Self {
+        match error {
+            StorageError::AlreadyExists => DeltaTransactionError::VersionExists,
+            _ => DeltaTransactionError::Storage { source: error },
+        }
+    }
+}
+
+pub struct DeltaTransactionOptions {
+    max_retry_commit_attempts: u32,
+}
+
+const DEFAULT_DELTA_MAX_RETRY_COMMIT_ATTEMPTS: u32 = 10000000;
```

Comment:
It is in fact 10 million retries (not ms). Feels a bit on the large side. I chose this value since it is the default used in the reference implementation. Open to other considerations.

Reply:
I did go ahead and commit your suggested change fwiw.